fastlane/lib/fastlane/plugins/plugin_info_collector.rb

```diff
@@ -49,7 +55,20 @@ def plugin_name_valid?(name)
       /^[a-z0-9_]+$/ =~ name &&
         # Does not contain the words 'fastlane' or 'plugin' since those will become
         # part of the gem name
-        [/fastlane/, /plugin/].none? { |regex| regex =~ name }
+        [/fastlane/, /plugin/].none? { |regex| regex =~ name } &&
+        # Plugin name isn't taken on RubyGems yet
+        !plugin_name_taken?(name)
+    end
+
+    # Checks if the plugin name is still free on RubyGems
+    def plugin_name_taken?(name)
+      require 'open-uri'
+      require 'json'
+      url = "https://rubygems.org/api/v1/gems/#{name}.json"
+      response = JSON.parse(open(url).read)
+      return !!response['version']
+    rescue
+      false
```

Comment:
It might make sense to memoize the result of this call for each name, so that we can avoid making the web request twice between `plugin_name_valid?` and `collect_plugin_name`